### PR TITLE
Python Arrow How-To-Guides

### PR DESCRIPTION
**Pull request by Alex-Monahan**
Hi Folks!

Let me know what you think! I followed the Pandas guides as an example. 

I didn't even know about Arrow Scanners, so that's worth knowing! Quick random question for my own learning: any idea on the speed of using an Arrow Scanner vs. just querying a dataset directly with DuckDB using a where clause? Arrow mentions that they use Async IO, so I didn't know if that would have any impact.

### CODE REVIEW

File: docs/guides/python/sql_on_arrow.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+layout: docu
+title: SQL on Apache Arrow
+selected: SQL on Apache Arrow
+---
+
+# How to execute SQL on Apache Arrow
+DuckDB can query multiple different types of Apache Arrow objects. 
+
```
Comment (pdet) [inline docs/guides/python/sql_on_arrow.md:9]: For completion maybe also add an example with recordbatchreaders?

File: docs/guides/python/export_arrow.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# connect to an in-memory database
+con = duckdb.connect()
+
+my_arrow_table = pa.Table.from_pydict({'i':[1,2,3,4],
+                                       'j':["one", "two", "three", "four"]})
+
+# query the Apache Arrow Table "my_arrow_table" and return as an Arrow Table
+results = con.execute("SELECT * FROM my_arrow_table").arrow()
+```
```
[the remark this answers is not in the document]
Comment (Alex-Monahan) [reply docs/guides/python/export_arrow.md:23]: Thanks for the feedback- I've added a section for that in the latest commit!

File: docs/guides/python/sql_on_arrow.md (hunk 1 above)
Comment (Alex-Monahan) [reply docs/guides/python/sql_on_arrow.md:9]: I also added an example for this! I'm learning a fair bit about Arrow... they have a few different options for objects, huh? I'm sure you feel that more than I do...

**Comment by Alex-Monahan** [comment]
I believe this is ready for another review @pdet! Thanks again!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**